nc-news/src/components/ArticleListByTopic.jsx

```diff
@@ -1,53 +1,47 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
 import * as api from "../api";
 
-function ArticlesListByTopic() {
+function ArticleListByTopic() {
+  const { topic } = useParams();
   const [articles, setArticles] = useState([]);
-  const [topics, setTopics] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const { topic } = useParams();
 
   useEffect(() => {
     setIsLoading(true);
     api.getArticles().then((articlesAPI) => {
       setArticles(articlesAPI);
       setIsLoading(false);
     });
-  }, []);
+  }, [topic]);
 
-  useEffect(() => {
-    setIsLoading(true);
-    api.getTopics().then((topicsAPI) => {
-      setTopics(topicsAPI);
-      setIsLoading(false);
-    });
-  }, []);
-  ////////////
   let articlesByTopic = articles.filter((article) => {
-    if (article.author === user.username) {
+    if (article.topic === topic) {
       return article;
     }
   });

```

Comment:
you should not be using a JavaScript filter here!

Getting the articles by the correct topic should be handled by your back end!

Have a look at what queries your `GET /api/articles` supports